File: Carnicus/Assets/NetworkModels/Responses/CharacterInfoResponse.cs

using System;
using Newtonsoft.Json;

public class CharacterInfoResponse : ModelInterface
{
	public string type = "CharacterInfoResponse";
	public string name;
	public float posx;
	public float posy;
	public float posz;
	public int GUID;
	
	public CharacterInfoResponse (string name, float posx, float posy, float posz, int GUID)
	{
		this.name = name;
		this.posx = posx;
		this.posy = posy;
		this.posz = posz;
		this.GUID = GUID;
	}
	
	public override string ToNetworkString()
	{
		return JsonConvert.SerializeObject(this);
	}
	
	public static CharacterInfoResponse ToModel(string networkString)
	{
		CharacterInfoResponse req = JsonConvert.DeserializeObject<CharacterInfoResponse>(networkString);
		if (req != null && req.type == "CharacterInfoResponse")
			return req;
		return null;
	}
}

File: CarnicusServer/src/main/java/com/malow/carnicusserver/models/responses/CharacterInfoResponse.java
package com.malow.carnicusserver.models.responses;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.malow.carnicusserver.models.ModelInterface;

import java.io.IOException;

public class CharacterInfoResponse implements ModelInterface
{
    public final String name;
    public final float posx;
    public final float posy;
    public final float posz;
    public final long GUID;

    @JsonCreator
    public CharacterInfoResponse(@JsonProperty("name") String name,
    							 @JsonProperty("posx") float posx,
    							 @JsonProperty("posy") float posy,
    							 @JsonProperty("posz") float posz,
    							 @JsonProperty("GUID") long GUID)
    {
        this.name = name;
        this.posx = posx;
        this.posy = posy;
        this.posz = posz;
        this.GUID = GUID;
    }

    @Override
    public String toNetworkString()
    {
        ObjectMapper mapper = new ObjectMapper();
        String str = "";
        try
        {
             str = mapper.writeValueAsString(this);
        } catch (IOException e)
        {
            e.printStackTrace();
        }
        return str;
    }
}
